tests/pubsub_logging_test.py

```diff
@@ -41,6 +55,77 @@ def test_compat_urlsafe_b64encode(self):
         self.assertEqual(expected, result)
 
 
+class GetPubsubClientTest(unittest.TestCase):
+    """Tests for utils.get_pubsub_client function.
+
+    You have to set GOOGLE_APPLICATION_CREDENTIALS pointing to the
+    json file of the service account.
+    """
+    RETRY = 3
+
+    def test_get_pubsub_client_with_service_account(self):
+        """Tests the client obtained by the service account method."""
+        client = get_pubsub_client()
+        project = os.environ.get(TEST_PROJECT_ENV, DEFAULT_TEST_PROJECT)
+        client.projects().topics().list(project=project).execute(
+            num_retries=self.RETRY)
+        # Providing an Http object this time.
+        client = get_pubsub_client(http=httplib2.Http())
+        project = os.environ.get(TEST_PROJECT_ENV, DEFAULT_TEST_PROJECT)
+        client.projects().topics().list(project=project).execute(
+            num_retries=self.RETRY)
+
+    def test_get_pubsub_client_with_scoped_credentials(self):
+        """Tests the client obtained by scoped credentials."""
+        credentials = GoogleCredentials.get_application_default()
+        credentials = credentials.create_scoped(PUBSUB_SCOPES)
+        client = get_pubsub_client(credentials=credentials)
+        project = os.environ.get(TEST_PROJECT_ENV, DEFAULT_TEST_PROJECT)
+        client.projects().topics().list(project=project).execute(
+            num_retries=self.RETRY)
+
+
+class CheckTopicTest(unittest.TestCase):
```

Comment:
Can we just make this an integration test as well? Create a topic in setUp if necessary, then check for it or some TOPIC_DOES_NOT_EXIST one in the tests.